### PR TITLE
Slack integration

### PR DESCRIPTION
**Pull request by TimCPB**

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -16,3 +17,29 @@ export const createMongoNoteInstance = async (
...
 };
+
+const returnUrlParameters = (url: string) => {
+  const splitUrl = url.split("/");
+  const index = splitUrl.length;
+  return [splitUrl[index - 3], splitUrl[index - 2], splitUrl[index - 1]];
+};
+
+export const slackQuery = (request: Request, response: Response) => {
+  const { body } = request;
```
Comment (benfurber) [inline src/utils.ts]: This can be:
```
   const { response_url } = request.body;
   ```

File: src/utils.ts [hunk 2]
```diff
@@ -16,3 +17,29 @@ export const createMongoNoteInstance = async (
...
+  if (body.response_url) {
+    const [param1, param2, param3] = returnUrlParameters(body.response_url);
+
+    console.log(request.body);
```
Comment (benfurber) [inline src/utils.ts]: Ideally nothing like this gets through to the main branch.

File: src/utils.ts [hunk 3]
```diff
@@ -16,3 +17,29 @@ export const createMongoNoteInstance = async (
...
+    response.send({
+      blocks: [
+        {
+          type: "section",
+          text: {
+            type: "mrkdwn",
+            text: `Hey <@${body.user_id}>\n<http://localhost:3000/?chan=${body.channel_id}&sender=${body.user_id}&p1=${param1}&p2=${param2}&p3=${param3}|Tape your message here!>\n:loud_sound:`,
```
Comment (benfurber) [inline src/utils.ts]: This line is too long.

File: src/utils.ts [hunk 4]
```diff
@@ -16,3 +17,29 @@ export const createMongoNoteInstance = async (
     }).save();
   }
 };
+
+const returnUrlParameters = (url: string) => {
+  const splitUrl = url.split("/");
+  const index = splitUrl.length;
+  return [splitUrl[index - 3], splitUrl[index - 2], splitUrl[index - 1]];
+};
```
Comment (benfurber) [inline src/utils.ts:25]: This can go into a util file.

File: src/index.ts [hunk 5]
```diff
@@ -1,13 +1,21 @@
...
 import { MONGO_URL } from "./constants";
 import resolvers from "./resolvers";
 import typeDefs from "./schema";
+import { slackQuery } from "./utils";
 
 const startServer = async () => {
   const app = express();
+  app.use(bodyParser.urlencoded({ extended: true }));
+
+  // may need to change to GraphQL query
```
Comment (benfurber) [inline src/index.ts]: Remove.

File: src/utils.ts [hunk 6]
```diff
@@ -16,3 +17,29 @@ export const createMongoNoteInstance = async (
...
+
+export const slackQuery = (request: Request, response: Response) => {
+  const { body } = request;
+  if (body.response_url) {
+    const [param1, param2, param3] = returnUrlParameters(body.response_url);
```
Comment (benfurber) [inline src/utils.ts]: I'd prefer that these names were more meaningful, but seeing as I know we're going to change them soon, don't worry about it.

**Review by benfurber — CHANGES_REQUESTED** [review]

File: src/utils.ts (hunk 4 above)
Comment (TimCPB) [reply src/utils.ts:25]: Do you want us to create a utils folder containing separate utils files? This is already in our utils file...

File: src/index.ts (hunk 5 above)
Comment (TimCPB) [reply src/index.ts]: I think we need line 13 in order to parse Slack's request, but we've replaced it with the updated version:
app.use(express.urlencoded({ extended: true }))

**Pull request merged**